### DISCUSSION TITLE
How can i know my wallet    connect or dissconnect by using SignClient

### DISCUSSION THREAD

**Discussion by arbaz-meo** [WalletConnect/walletconnect-monorepo#2567] [Developer Support] [answered]
if i connect my wallet through signclient then how can i know my wallet dissconnect or still connected by using topic . we doing all the things on the server and on server we not using any modal . the same question i have asked on discord but i not got solution yet .

**Comment by arbaz-meo** [comment]
@finessevanes , @0xAsimetriq  can you give me solution of this  ??

**Comment by glitch-txs** [comment]
You can use event listener

https://docs.walletconnect.com/2.0/web/core/pairing-api#listeners-for-pairing-related-events

**Reply by arbaz-meo** [reply]
@glitch-txs thats listener only applicable if session delete by user i want to check my wallet connect or not by using topic

**Reply by glitch-txs** [reply, marked as answer]
Ah I see, something like this right?

https://github.com/WalletConnect/walletconnect-monorepo/discussions/2511#discussioncomment-6101860

**Comment by finessevanes** [comment]
@arbaz-meo @glitch-txs suggestion should help you achieve what you're looking for.

**Reply by arbaz-meo** [reply]
@finessevanes yeah its solution  for right now but i will request to have some function in next version which can give us update against the specific  topic
